app/javascript/src/components/Player.tsx

```diff
@@ -720,10 +707,8 @@ class Player extends React.Component<PlayerProps, PlayerState> {
 
   private debugRerender() {
     this.setState({
-      focus: this.state.focus,
       currentItems: this.state.currentItems,
-      currentStats: this.state.currentStats,
-      history: this.state.history
+      currentStats: this.state.currentStats
```

Comment:
I think what you want to do is assign a new set of items to `state.currentItems`. 

Mutating `this.state` can lead to bugs and is generally not a supported pattern. 

I would rewrite [the `removeItem` callback](https://github.com/vigetlabs/storyboard/blob/8c2f8c4a6423b2998977e4c552363a139baa6ac4/app/javascript/src/components/Player.tsx#L716-L724) so that it does not mutate `this.state.currentItems` (`this.state.currentItems = newItems`). You're already cloning the list to make your change, so I think you can just replace that line ^ with `this.setState({ currentItems: newItems })` and get rid of `debugRerender` altogether.

Happy to chat through it if you have questions!